apps/modernization-ui/src/SkipLink/SkipLinkContext.tsx

```diff
@@ -0,0 +1,37 @@
+import React, { createContext, useContext, useState } from 'react';
+
+interface SkipLinkContextType {
+    setId: (id: string) => void;
+}
+
+// Create the context
+export const SkipLinkContext = createContext<SkipLinkContextType | undefined>(undefined);
+
+// Custom hook for the context
+export function useSkipLink() {
+    const context = useContext(SkipLinkContext);
+    if (context === undefined) {
+        throw new Error('useSkipLink must be used within a SkipLinkProvider');
+    }
+    return context;
+}
+
+interface SkipLinkProviderProps {
+    children: React.ReactNode;
+}
+
+export const SkipLinkProvider: React.FC<SkipLinkProviderProps> = ({ children }: SkipLinkProviderProps) => {
+    const [id, setId] = useState<string | undefined>(undefined);
+
+    // Setting up the id to pass it to the anchor tag
+    const contextValue: SkipLinkContextType = {
+        setId
+    };
+
+    return (
+        <SkipLinkContext.Provider value={contextValue}>
+            <a href={'#' + id}>Skip to main content</a>
```

Comment:
suggestion:  only display the skip link if `id` exists to avoid any possible confusion when the link doesn't actually go anywhere

Reply:
okay will add the check